.github/workflows/msbuild.yml

```diff
@@ -66,3 +71,65 @@ jobs:
       # This is the best way to fix an issue in master that was only caught by the nightly build.
       all_platforms: ${{ contains(github.event.pull_request.labels.*.name, 'run-deep-tests') || contains(github.event.push.labels.*.name, 'run-deep-tests')}}
       num_shards: 5
+
+  test-coverage-analysis:
+    runs-on: ubuntu-20.04
+    needs: [xunit-tests, integration-tests]
+    steps:
+      # Check out Dafny so that highlighted source is possible
+      - name: Checkout Dafny
+        uses: actions/checkout@v3
+        with:
+          path: dafny
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{env.dotnet-version}}
+
+      - name: Install dotnet-reportgenerator and coverage
+        run: |
+          dotnet tool install --global dotnet-reportgenerator-globaltool
+          dotnet tool install --global dotnet-coverage
+
+      - name: Download integration test artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: integration-test-results-ubuntu-20.04
+
+      - name: Download unit test artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: unit-test-results-ubuntu-20.04
+
+      - name: Show files
```

Comment:
This looks like debugging code. Do you want it long term?

Reply:
Probably not, now that you mention it. :)